src/core/entity.js

```diff
@@ -186,58 +175,50 @@ class Entity {
     mat4.translate(this.matrix, this.matrix, move);
   }
 
-  move_r(dist) {
-    this.move_along(vec3.left, dist);
-  }
-
-  move_u(dist) {
-    this.move_along(vec3.up, dist);
-  }
-
-  move_f(dist) {
-    this.move_along(vec3.forward, dist);
-  }
-
-  do_step(tick_length) {
-    if (this.dir.f && !this.dir.b) {
-      this.move_f(tick_length / 1000 * this.move_speed);
+  do_move(tick_length, current_dir) {
+    if (current_dir.f) {
+      this.move_along(vec3.forward, tick_length / 1000 * this.move_speed);
```

Comment:
Since this is duplicated multiple times among this method, maybe it would be smart to cache `tick_length / 1000 * this.move_speed` value? It's possible the entity will move in multiple direction in the same frame/method call.